components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.spec.ts

```diff
@@ -0,0 +1,177 @@
+      expect(component.firstFormGroup.controls['description'].value).toEqual('');
+      expect(component.thirdFormGroup.controls['dattr'].value).toEqual('{}');
+      expect(component.fourthFormGroup.controls['oattr'].value).toEqual('{}');
+    });
+
+    it('on conflict error, modal remains open and displays conflict error', () => {
+      spyOn(component.conflictErrorEvent, 'emit');
+      component.visible = true;
+      component.firstFormGroup.controls['name'].setValue(environment.name);
+      component.firstFormGroup.controls['description'].setValue(environment.description);
+      component.constraintArray = environment.cookbook_versions;
+      component.thirdFormGroup.controls['dattr'].setValue(
+        JSON.stringify(environment.default_attributes));
+      component.fourthFormGroup.controls['oattr'].setValue(
+        JSON.stringify(environment.override_attributes));
+
+      component.createEnvironment();
+
+      const conflict = <HttpErrorResponse>{
+        status: HttpStatus.CONFLICT,
+        ok: false
+      };
+      store.dispatch(new CreateEnvironmentFailure(conflict));
+      expect(component.visible).toBe(true);
+    });
+
+    it('on success, closes modal and adds new environment', () => {
+      spyOn(component.conflictErrorEvent, 'emit');
+      component.visible = true;
+      component.firstFormGroup.controls['name'].setValue(environment.name);
+      component.firstFormGroup.controls['description'].setValue(environment.description);
+      component.constraintArray = environment.cookbook_versions;
+      component.thirdFormGroup.controls['dattr'].setValue(
+        JSON.stringify(environment.default_attributes));
+      component.fourthFormGroup.controls['oattr'].setValue(
+        JSON.stringify(environment.override_attributes));
+      component.createEnvironment();
+
+      store.dispatch(new CreateEnvironmentSuccess({environment}));
+      component.closeCreateModal();
+      expect(component.environmentsList.length).not.toBeNull();
+    });
+
+    it('on create error, modal is closed (because error is handled by failure banner)', () => {
+      spyOn(component.conflictErrorEvent, 'emit');
+      component.visible = true;
+      component.firstFormGroup.controls['name'].setValue(environment.name);
+      component.firstFormGroup.controls['description'].setValue(environment.description);
+      component.constraintArray = environment.cookbook_versions;
+      component.thirdFormGroup.controls['dattr'].setValue(
+        JSON.stringify(environment.default_attributes));
+      component.fourthFormGroup.controls['oattr'].setValue(
+        JSON.stringify(environment.override_attributes));
+
+      component.createEnvironment();
+
+      const error = <HttpErrorResponse>{
+        status: HttpStatus.INTERNAL_SERVER_ERROR,
+        ok: false
+      };
+
+      store.dispatch(new CreateEnvironmentFailure(error));
+
+      expect(component.creating).toBe(false);
```

Comment:
An earlier test confirms the visibility of the modal with `component.visible`; here you are using  `component.creating`. Why are they not testing the same property?

Reply:
In code, getting the status from the store we are setting the creating to false therefore I have set it to false in spec. Now I have updated the changes.